File: frontend/ios/SiriRN.m
//
//  SiriRN.m
//  frontend
//
//  Created by Domiziano Scarcelli on 02/04/23.
//

#import <Foundation/Foundation.h>
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(SiriRN, NSObject)
  RCT_EXTERN_METHOD(handleNearbyRestaurantsIntent: (RCTResponseSenderBlock)callback)
@end

File: frontend/ios/SiriRN.swift
import Foundation

@objc(SiriRN)
public class SiriRN: NSObject {
  
//   init(message: String) {
//     self.message = message
//   }

  private var message = "Daje roma daje"

  @objc
  func handleNearbyRestaurantsIntent(_ callback: RCTResponseSenderBlock){
    callback([message])
  }

  @objc
  static func requiresMainQueueSetup() -> Bool {
    return true
  };

  @objc
  func constantsToExport() -> [String: Any] {
    return ["initialMessage": ""]
  }
}
